sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -96,6 +96,30 @@ case class CatalogStorageFormat(
     }
     map
   }
+
+  /** Util providing the JSON-compatible version of toLinkedHashMap, for serialization of fields
```

Comment:
I don't quite understand what JSON-compatible means. All string values will become JSON string, and looking at the code we just add more `"`, so the final JSON becomes `"location": "\"/my/path\""`?

Reply:
I think if we don't add the additional `"..."` then raw string values won't be parsed as a string by json4s. It doesn't come out as double-quotes in the output (e.g. [golden file](https://github.com/apache/spark/blob/b85838215f9cb878ae0becb41bc6a0d61963e0c3/sql/core/src/test/resources/sql-tests/results/describe.sql.out)